task160_FinalTask/src/test/java/WebDriverSingleton/WebDriverSingleton.java

```diff
@@ -0,0 +1,37 @@
+package WebDriverSingleton;
+
+
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.openqa.selenium.firefox.FirefoxDriver;
+
+import java.util.concurrent.TimeUnit;
+
+public class WebDriverSingleton {
+    private static WebDriverSingleton object = null;
+    private WebDriver driver;
```

Comment:
It will work for one-thread execution only, you need to use ThreadLocal for multi-threading

Reply:
What were my mistakes:
1. At first I was trying to make the only WebDriver as ThreadLocal, but the singleton object was still dependent. And every 3rd test(if there were 2 threads) brokes this logic.
2. When I've trying to make Singleton object, I've made a mistake and compare "object == null" and always receive NullPointerException error
3. And finally I've changed this rule on object.get() == null and it starts working.